jdaviz/configs/imviz/plugins/orientation/orientation.py

```diff
@@ -392,8 +413,8 @@ def _change_reference_data(self, *args, **kwargs):
                 self.orientation.selected, viewer_id=self.viewer.selected
             )
 
-    def _on_refdata_change(self, msg={}):
-        self.orientation.only_wcs_layers = msg.data.meta.get('_WCS_ONLY', False)
+    def _on_refdata_change(self, msg):
+        self.orientation.set_wcs_only_filter(wcs_only=msg.data.meta.get('_WCS_ONLY', False))
         if hasattr(self, 'viewer'):
             ref_data = self.ref_data
             viewer = self.app.get_viewer(self.viewer.selected)

```

Comment:
Using`self.viewer.selected_obj` here will be slightly more efficient since it will cache